source/initial-dns-seedlist-discovery/tests/README.rst

```diff
@@ -114,31 +114,49 @@ These YAML and JSON files contain the following fields:
-client's initial seedlist matches the list in ``seeds``. If ``numSeeds`` is
-specified, drivers SHOULD verify that the size of that set matches ``numSeeds``.
-
-If ``hosts`` is specified, drivers MUST verify that the set of
-ServerDescriptions in the client's TopologyDescription eventually matches the
-list in ``hosts``. If ``numHosts`` is specified, drivers MUST verify that the
-size of that set matches ``numHosts``.
-
-If ``options`` is specified, drivers MUST verify each of the values under
-``options`` match the MongoClient's parsed value for that option. There may be
-other options parsed by the MongoClient as well, which a test does not verify.
-
-If ``parsed_options`` is specified, drivers MUST verify that each of the values
-under ``parsed_options`` match the MongoClient's parsed value for that option.
-Supported values include, but are not limited to, ``user`` and ``password``
-(parsed from ``UserInfo``) and ``auth_database`` (parsed from
-``Auth database``).
-
-If ``error`` is specified and ``true``, drivers MUST verify that an error has
-been thrown.
+For each YAML file:
+
+- Create a MongoClient initialized with the ``mongodb+srv``
+  connection string.
+- Run run a "ping" operation if ``ping`` in the test file is true.
+
+Assertions:
+
+- If ``seeds`` is specified, drivers SHOULD verify that the set of hosts in the
+  client's initial seedlist matches the list in ``seeds``. If ``numSeeds`` is
+  specified, drivers SHOULD verify that the size of that set matches
+  ``numSeeds``.
+
+- If ``hosts`` is specified, drivers MUST verify that the set of
+  ServerDescriptions in the client's TopologyDescription eventually matches the
+  list in ``hosts``. If ``numHosts`` is specified, drivers MUST verify that the
+  size of that set matches ``numHosts``.
+
+- If ``options`` is specified, drivers MUST verify each of the values under
+  ``options`` match the MongoClient's parsed value for that option. There may be
+  other options parsed by the MongoClient as well, which a test does not verify.
+
+- If ``parsed_options`` is specified, drivers MUST verify that each of the
+  values under ``parsed_options`` match the MongoClient's parsed value for that
+  option. Supported values include, but are not limited to, ``user`` and
+  ``password`` (parsed from ``UserInfo``) and ``auth_database`` (parsed from
+  ``Auth database``).
+
+- If ``error`` is specified and ``true``, drivers MUST verify that initializing
+  the MongoClient throws an error. If ``error`` is not specified or is
+  ``false``, both initializing the MongoClient and running a ping operation must
+  succeed without throwing any errors.
+
+- If ``ping`` is specified and ``true``, drivers MUST verify that running a
+  "ping" operation using the initialized MongoClient succeeds. If ``ping`` is
+  not specified or is ``false``, drivers MUST NOT run a "ping" operation.
+  **Note:** These tests are expected to be run against MongoDB databases with
```

Comment:
can we update this text in light of the changes you had to make around tests with incorrect auth info?

Reply:
Extended the note to clarify the expected "ping" behavior when userinfo is missing vs when userinfo is incorrect.